crates/sui-core/src/authority/authority_store.rs

```diff
@@ -398,11 +398,11 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
     }
 
     /// Read a transaction envelope via lock or returns Err(TransactionLockDoesNotExist) if the lock does not exist.
-    pub async fn get_transaction_envelope(
+    pub async fn get_transaction_lock(
```

Comment:
This shouldn't have been renamed.  Clearly, what is being returned is the `TransactionEnvelope`, it does not return a lock.

Reply:
I kind of does, though.
A function called `get_transaction_envelope` in a store will make you think that it would return you a transaction given a digest.
But here we are requesting the transaction that locks a given object. Maybe `get_object_locking_transaction`?

Reply:
How about `get_transaction_from_lock`?  or `get_transaction_from_locked_obj`?   I think that's closer.